tasks.py

```diff
@@ -34,35 +34,44 @@ def create(name, filename="tasks.csv"):
     Create a new task.
     """
 
-    with open(filename, "a") as tasks_file:
+    with open(filename, "a", newline="") as tasks_file:
         writer = csv.writer(tasks_file)
+        print(name)
         writer.writerow([name, False])
 
 
-def complete():
+def complete(stdout=sys.stdout,filename="tasks.csv"):
     """
     Mark an existing task as completed.
     """
 
     with (
-        open("tasks.csv") as tasks_file,
-        NamedTemporaryFile("w", delete=False) as new,
-    ):
+        open(filename) as tasks_file,
+        NamedTemporaryFile("w", delete=False) as new):
         reader = csv.reader(tasks_file)
         print("Current tasks:")
         for id, (name, completed) in enumerate(reader):
-            print(id, name, completed)
+            print(id+1, name, DONE if eval(completed) else '')
 
         to_complete = int(input("task ID?> "))
-        writer = csv.writer(new)
+        writer = csv.writer(new, lineterminator="\n")
         tasks_file.seek(0)
         for id, (name, completed) in enumerate(reader):
-            if id == to_complete:
+
+            if id+1 == to_complete and eval(completed) == True:
+
+                try:
```

Comment:
I don't follow the `try/except` here, this is just equivalent to always calling `stdout.write("Already completed, try again!\n")` because you are always raising an exception and then immediately catch it.